Спринт 10

NastasyaRosh opened the pull request:
-refactor существующий ранее код разбит на хранилища (service, controller и тп) -refactor добавлены зависимости через spring
-feat добавлена функция добавления друзей и постановки лайков фильмам

NastasyaRosh committed:
Спринт 10
-refactor существующий ранее код разбит на хранилища (service, controller и тп)
-refactor добавлены зависимости через spring
-feat добавлена функция добавления друзей и постановки лайков фильмам

NastasyaRosh committed:
Спринт 10
-refactor существующий ранее код разбит на хранилища (service, controller и тп)
-refactor добавлены зависимости через spring
-feat добавлена функция добавления друзей и постановки лайков фильмам

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,63 @@
+
+    public List<Film> popularFilms(Integer count) {
+        if (count == null) {
+            count = 10;
```
Лучше стараться числа выносить из кода в константы, переменные и так далее. Плюс дав информативное название константе, вы помогаете читающему понять, какой смысл у этого числа.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,63 @@
+        if (count == null) {
+            count = 10;
+        }
+        return filmStorage.allFilms().stream().sorted((f1, f2) -> (f2.getLikes().size() - f1.getLikes().size())).limit(count).collect(Collectors.toList());
```
Стараются ограничивать длину строки не более 120 символов. И если больше, то переносят на следующую. Для юзеров аналогично.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,64 @@
+
+    @Override
+    public Film addFilm(Film film) {
+        validate(film);
```
Валидация должна быть в сервисе, а не в слое хранения данных.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,70 @@
+
+    @Override
+    public User addUser(User user) {
+        forNullName(user);
```
Лучше перенести в сервисе эту операцию.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,70 @@
+    @Override
+    public User addUser(User user) {
+        forNullName(user);
+        if (checkEmail(user.getEmail()).isPresent()) {
```
Валидацию тоже в сервис.

NastasyaRosh committed:
Спринт 10.2
-style значение DEFAULT_COUNT вынесено в константу из метода
-style добавлены переносы строк там, где длина строки превышала 120 символов
-refactor все валидации вынесены из InMemoryStorage в Service (и для User и для Film)

NastasyaRosh commented:
Спасибо за комментарии :)
Направляю исправленную версию.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,79 @@
+            count = DEFAULT_COUNT;
+        }
+        return filmStorage.allFilms().stream().
+                sorted((f1, f2) -> (f2.getLikes().size() - f1.getLikes().size())).
```
Дело вкуса, но чаще встречаю написание, когда строка с методом начинается с точки.
.sorted()
.limit()
.collect()

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,60 @@
+import ru.yandex.practicum.filmorate.exception.FilmOrUserNotExist;
+import ru.yandex.practicum.filmorate.model.User;
+
+import java.util.*;
```
На мой взгляд, лучше явно прописывать классы, которые импортируются.
